source/documentation/fuseki2/fuseki-configuration.md

```diff
@@ -124,9 +124,71 @@ An in-memory dataset, with data in the default graph taken from a local file.
         ## tdb:unionDefaultGraph true ;
          .
 
+#### TDB2
+
+    <#dataset> rdf:type      tdb:DatasetTDB2 ;
+        tdb:location "/some/path" ;
+        # this can also have unionDefaultGraph, etc.
+         .
+
 #### Inference
 
-> @@
+An inference reasoner can be layered on top of a dataset as defined above. The type of reasoner must be selected carefully and should not include more reasoning than is required by the application, as extensive reasoning can be detrimental to performance.
+
+You have to build up layers of dataset, inference model, and graph.
+
+    <#dataset> rdf:type ja:RDFDataset;
+         ja:defaultGraph <#inferenceModel>
+         .
+         
+    <#inferenceModel> rdf:type      ja:InfModel;
+         ja:reasoner [ ja:reasonerURL <http://example/someReasonerURLHere> ];
+         ja:baseModel <#baseModel>;
+         .
+    <#baseModel> rdf:type tdb:GraphTDB2;  # for example.
+         tdb2:location "/some/path/to/store/data/to";
+         # etc
+         .
+
+
+##### Possible reasoners:
+
+* **Generic Rule Reasoner**: `http://jena.hpl.hp.com/2003/GenericRuleReasoner`
+  
+  The specific rule set and mode configuration can be set either be method calls to the created reasoner or though parameters in the configuration Model.
```

Comment:
"can be set either be method"... by method?

"or though"... or through? (not a native speaker, so take it with a grain of salt)